modules/service/src/main/scala/lucuma/itc/input/customSed/CustomSedDatResolver.scala

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.itc.input.customSed
+
+import cats.Order.given
+import cats.data.NonEmptyMap
+import cats.effect.Concurrent
+import cats.syntax.all.*
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import lucuma.core.math.Wavelength
+
+import scala.collection.immutable.SortedMap
+
+/**
+ * Implementation of `CustomSed.Resolver` that parses a `dat` file, consisting of lines of the form
+ * "`wavelength density`".
+ *
+ * This trait just provides the parsing functionality, and does not specify how to obtain the `dat`
+ * file.
+ */
+trait CustomSedDatResolver[F[_]: Concurrent] extends CustomSed.Resolver[F]:
```

Comment:
Do we have teests for the parser?

Reply:
It is tested as part of `GraphQLCustomSedSuite`.